diff -up xpdf-3.03/goo/gmem.cc.alloc xpdf-3.03/goo/gmem.cc
--- xpdf-3.03/goo/gmem.cc.alloc	2011-08-15 17:08:53.000000000 -0400
+++ xpdf-3.03/goo/gmem.cc	2011-08-22 13:48:31.891115878 -0400
@@ -47,9 +47,9 @@ static int gMemInUse = 0;
 
 #endif /* DEBUG_MEM */
 
-void *gmalloc(int size) GMEM_EXCEP {
+void *gmalloc(size_t size) GMEM_EXCEP {
 #ifdef DEBUG_MEM
-  int size1;
+  size_t size1;
   char *mem;
   GMemHdr *hdr;
   void *data;
@@ -122,11 +122,11 @@ void *gmalloc(int size) GMEM_EXCEP {
 #endif
 }
 
-void *grealloc(void *p, int size) GMEM_EXCEP {
+void *grealloc(void *p, size_t size) GMEM_EXCEP {
 #ifdef DEBUG_MEM
   GMemHdr *hdr;
   void *q;
-  int oldSize;
+  size_t oldSize;
 
   if (size < 0) {
 #if USE_EXCEPTIONS
@@ -227,7 +227,7 @@ void *greallocn(void *p, int nObjs, int 
 
 void gfree(void *p) {
 #ifdef DEBUG_MEM
-  int size;
+  size_t size;
   GMemHdr *hdr;
   unsigned long *trl, *clr;
 
diff -up xpdf-3.03/goo/gmem.h.alloc xpdf-3.03/goo/gmem.h
--- xpdf-3.03/goo/gmem.h.alloc	2011-08-15 17:08:53.000000000 -0400
+++ xpdf-3.03/goo/gmem.h	2011-08-22 13:47:27.274853540 -0400
@@ -36,13 +36,13 @@ extern "C" {
  * Same as malloc, but prints error message and exits if malloc()
  * returns NULL.
  */
-extern void *gmalloc(int size) GMEM_EXCEP;
+extern void *gmalloc(size_t size) GMEM_EXCEP;
 
 /*
  * Same as realloc, but prints error message and exits if realloc()
  * returns NULL.  If <p> is NULL, calls malloc instead of realloc().
  */
-extern void *grealloc(void *p, int size) GMEM_EXCEP;
+extern void *grealloc(void *p, size_t size) GMEM_EXCEP;
 
 /*
  * These are similar to gmalloc and grealloc, but take an object count
