--- ./vboxnetflt/linux/VBoxNetFlt-linux.c.orig	2015-04-22 09:22:38.700928942 +0200
+++ ./vboxnetflt/linux/VBoxNetFlt-linux.c	2015-04-22 09:24:39.313927297 +0200
@@ -889,7 +889,7 @@
         pBuf->mac_len = uMacLen;
 # if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 27)
         /* Restore VLAN tag stripped by host hardware */
-        if (vlan_tx_tag_present(pBuf) && skb_headroom(pBuf) >= VLAN_ETH_HLEN)
+        if (skb_vlan_tag_present(pBuf) && skb_headroom(pBuf) >= VLAN_ETH_HLEN)
         {
             uint8_t *pMac = (uint8_t*)skb_mac_header(pBuf);
             struct vlan_ethhdr *pVHdr = (struct vlan_ethhdr *)(pMac - VLAN_HLEN);
@@ -899,7 +899,7 @@
             memmove(pVHdr, pMac, VLAN_ETH_ALEN * 2);
 #  endif
             pVHdr->h_vlan_proto = RT_H2N_U16(ETH_P_8021Q);
-            pVHdr->h_vlan_TCI   = RT_H2N_U16(vlan_tx_tag_get(pBuf));
+            pVHdr->h_vlan_TCI   = RT_H2N_U16(skb_vlan_tag_get(pBuf));
             pBuf->mac_header   -= VLAN_HLEN;
             pBuf->mac_len      += VLAN_HLEN;
         }
