diff -uNr linux-3.13-orig/drivers/usb/host/xhci.c linux-3.13/drivers/usb/host/xhci.c
--- linux-3.13-orig/drivers/usb/host/xhci.c	2014-01-20 03:40:07.000000000 +0100
+++ linux-3.13/drivers/usb/host/xhci.c	2014-03-27 16:53:55.138987198 +0100
@@ -4713,8 +4713,8 @@
 	struct device		*dev = hcd->self.controller;
 	int			retval;
 
-	/* Accept arbitrarily long scatter-gather lists */
-	hcd->self.sg_tablesize = ~0;
+	/* Limit the block layer scatter-gather lists to half a segment. */
+	hcd->self.sg_tablesize = TRBS_PER_SEGMENT / 2;
 
 	/* support to build packet from discontinuous buffers */
 	hcd->self.no_sg_constraint = 1;
diff -uNr linux-3.13-orig/drivers/usb/host/xhci.h linux-3.13/drivers/usb/host/xhci.h
--- linux-3.13-orig/drivers/usb/host/xhci.h	2014-01-20 03:40:07.000000000 +0100
+++ linux-3.13/drivers/usb/host/xhci.h	2014-03-27 16:53:54.646987220 +0100
@@ -1260,7 +1260,7 @@
  * since the command ring is 64-byte aligned.
  * It must also be greater than 16.
  */
-#define TRBS_PER_SEGMENT	64
+#define TRBS_PER_SEGMENT	256
 /* Allow two commands + a link TRB, along with any reserved command TRBs */
 #define MAX_RSVD_CMD_TRBS	(TRBS_PER_SEGMENT - 3)
 #define TRB_SEGMENT_SIZE	(TRBS_PER_SEGMENT*16)
